tools/Tool.cc

```diff
@@ -51,6 +57,61 @@ static bool outputIsColor() {
     return sColor;
 }
 
+static bool tokenize(const char* input, deque<string> &args)
+{
+    string str(input);
+    int start = 0;
+    for( size_t i=0; i<str.length(); i++){
+        char c = str[i];
+        if( c == ' ' ) {
+            if(i - start > 1) {
+                args.push_back(str.substr(start, i - start));
+            }
+            
+            while(str[i] == ' ') {
+                i++;
+            }
+            
+            start = i;
+            i--;
+        } else if(c == '\"' ) {
+            i++;
+            while( i < str.size() - 1 && str[i] != '\"' ){ i++; }
+            if(str[i] != '\"') {
+                return false;
+            }
+            
+            args.push_back(str.substr(start + 1, i - start - 1));
+            start = i;
+        } else if(c == '\r' || c == '\n') {
+            break;
+        } else if(!isascii(c)) {
+            start = i+1;
+        }
+    }
+    
+    int offset = 0;
+    int carriageReturn = str.find("\r");
+    int lineFeed = str.find("\n");
+    if(carriageReturn > 0) {
+        offset++;
+    }
+    
+    if(lineFeed > 0) {
+        offset++;
+    }
+    
+    if(str[str.length() - offset - 1] == '\"') {
+        offset++;
+    }
+    
+    if(str.length() - start - offset > 1) {
+        args.push_back(str.substr(start, str.length() - start - offset));
```

Comment:
It would be a lot easier to handle this last argument as part of normal processing, if you scanned `input` until you hit a `\0`.